pkg/pillar/pubsub/publish.go

```diff
@@ -191,13 +191,16 @@ func (pub *PublicationImpl) populate() {
 
 	pairs, restarted, err := pub.driver.Load()
 	if err != nil {
-		log.Fatalf(err.Error())
+		// Could be a truncated or empty file
+		log.Error(err)
+		return
```

Comment:
Don't we want to repair it here? Copy the corrupted file out to some palace and create a dummy file?

Reply:
We could remove it, or move it to .bak (so it can be inspected if you can ssh in).
With this code it lives forever so you can inspect it.